build.gradle.kts

```diff
@@ -295,8 +296,91 @@ subprojects {
                 sign(publishing.publications["shadow"])
             }
         }
-    }
+    } else {
+        apply(plugin = "signing")
+        apply(plugin = "maven-publish")
+        apply(plugin = "net.linguica.maven-settings")
+
+        val sourcesJar by tasks.registering(Jar::class) {
+            archiveClassifier.set("sources")
+            from(sourceSets.main.get().allSource)
+        }
+
+        val javadocJar by tasks.registering(Jar::class) {
+            archiveClassifier.set("javadoc")
+            from(tasks["javadoc"])
+        }
+
+        publishing {
+            publications {
+                create<MavenPublication>("mavenJava") {
+                    artifact(sourcesJar.get())
+                    artifact(javadocJar.get())
+
+                    pom {
+                        name.set("Gradle Manipulation Extension")
+                        description.set("A tool to work with ProjectNCL to manipulate Gradle builds.")
+                        url.set("https://github.com/project-ncl/gradle-manipulator")
+                        packaging = "jar"
+
+                        licenses {
+                            license {
+                                name.set("The Apache License, Version 2.0")
+                                url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+                            }
+                        }
+                        developers {
+                            developer {
+                                id.set("geoand")
+                                name.set("Georgios Andrianakis")
+                                email.set("gandrian@redhat.com")
+                            }
+                            developer {
+                                id.set("rnc")
+                                name.set("Nick Cross")
+                                email.set("ncross@redhat.com")
+                            }
+                            developer {
+                                id.set("TomasHofman")
+                                name.set("Tomas Hofman")
+                                email.set("thofman@redhat.com")
+                            }
+                            developer {
+                                id.set("metacosm")
+                                name.set("Chris Laprun")
+                                email.set("claprun@redhat.com")
+                            }
+                        }
+                        scm {
+                            connection.set("scm:git:http://github.com/project-ncl/gradle-manipulator.git")
+                            developerConnection.set("scm:git:git@github.com:project-ncl/gradle-manipulator.git")
+                            url.set("https://github.com/project-ncl/gradle-manipulator")
+                        }
+                    }
+                }
+            }
```

Comment:
I am not seeing a common jar:
```
NCL-6109(dwalluck/NCL-6109)|…6⚑4] 🎩❯ l ~/.m2/repository/org/jboss/gm/common/2.8-SNAPSHOT                                                                               14:50
total 524
-rw-rw-r--. 1 rnc rnc 497518 May 24 14:45 common-2.8-SNAPSHOT-javadoc.jar
-rw-rw-r--. 1 rnc rnc   1592 May 24 14:45 common-2.8-SNAPSHOT.pom
-rw-rw-r--. 1 rnc rnc  25646 May 24 14:45 common-2.8-SNAPSHOT-sources.jar
-rw-rw-r--. 1 rnc rnc    942 May 24 14:45 maven-metadata-local.xml
```

While the local jar does look *much* better, I do wonder whether we can avoid the large amount of duplication here?

Reply:
Ok now I am seeing a jar.